core/server/master/src/main/java/alluxio/master/meta/UpdateCheck.java

```diff
@@ -84,6 +85,18 @@ public static String getLatestVersion(String clusterID, long connectionRequestTi
    */
   @VisibleForTesting
   public static String getUserAgentString(String clusterID) throws IOException {
+    Joiner joiner = Joiner.on("; ").skipNulls();
+    String sysInfo = joiner.join(getUserAgentEnvironmentString(clusterID), getFeatureString());
+    return String.format("Alluxio/%s (%s)", ProjectConstants.VERSION, sysInfo);
+  }
+
+  /**
+   * @param clusterID the cluster ID
+   * @return a string representation of the user's environment in the format "key1:value1, key2:
```

Comment:
is this javadoc accurate? i don't think they are always key/value pairs and each section is separated by `;`, not `,`

Reply:
addressed